shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/instance/InstanceContext.java

```diff
@@ -141,13 +144,13 @@ public void deleteComputeNodeInstance(final ComputeNodeInstance instance) {
      * @param labels collection of contained label
      * @return compute node instances
      */
-    public Collection<ComputeNodeInstance> getComputeNodeInstances(final InstanceType instanceType, final Collection<String> labels) {
-        Collection<ComputeNodeInstance> result = new ArrayList<>(computeNodeInstances.size());
-        computeNodeInstances.forEach(each -> {
-            if (each.getInstanceDefinition().getInstanceType() == instanceType && each.getLabels().stream().anyMatch(labels::contains)) {
-                result.add(each);
+    public List<InstanceId> getComputeNodeInstanceIds(final InstanceType instanceType, final Collection<String> labels) {
```

Comment:
Collection maybe better.

Reply:
> Collection maybe better.

@totalo This result needs to be passed to the load balancing algorithm for use, and the value needs to be obtained by index.

Reply:
Get it!